pkg/internal/tests/frozen_time.go

```diff
@@ -0,0 +1,63 @@
+package tests
+
+import (
+	"time"
+
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FrozenTime is an interface for testing out fake times with additional
+// methods for protobuf and Kubernetes time formats.
+type FrozenTime interface {
+	// Now never changes
+	Now() time.Time
+	Before(time.Duration) time.Time
+	After(time.Duration) time.Time
+	// Before Now() by int64 seconds
+	BeforeSec(int64) time.Time
+	// After Now() by int64 seconds
+	AfterSec(int64) time.Time
+
+	MetaV1Now() *metav1.Time
+	MetaV1Before(time.Duration) *metav1.Time
+	MetaV1After(time.Duration) *metav1.Time
+	MetaV1BeforeSec(int64) *metav1.Time
+	MetaV1AfterSec(int64) *metav1.Time
+
+	PbNow() *timestamppb.Timestamp
+	PbBefore(time.Duration) *timestamppb.Timestamp
+	PbAfter(time.Duration) *timestamppb.Timestamp
+	PbBeforeSec(int64) *timestamppb.Timestamp
+	PbAfterSec(int64) *timestamppb.Timestamp
+}
+
+func NewFrozenTimeUnix(unix int64) FrozenTime {
+	return &ft{t: time.Unix(unix, 0)}
+}
+
+func NewFrozenTime(t time.Time) FrozenTime {
+	return &ft{t}
+}
+
+type ft struct {
+	t time.Time
+}
+
+func (f *ft) Now() time.Time                   { return f.t }
+func (f *ft) Before(d time.Duration) time.Time { return f.Now().Add(d) }
+func (f *ft) After(d time.Duration) time.Time  { return f.Now().Add(-d) }
+func (f *ft) AfterSec(s int64) time.Time       { return f.Now().Add(time.Duration(s) * time.Second) }
+func (f *ft) BeforeSec(s int64) time.Time      { return f.Now().Add(time.Duration(-s) * time.Second) }
```

Comment:
nit: can you sort these like the other groups? After, Before, AfterSec, BeforeSec... for consistency

Reply:
Sorted in the order mentioned in the interface